celeri/hmatrix.py

```diff
@@ -410,7 +413,15 @@ def build_hmatrix(
         U, S, V = np.linalg.svd(h_block, full_matrices=False)
         block_tol = tol * (h_block.size / reshaped.size) * frob_est
         frob_K = np.sqrt(np.cumsum(S[::-1] ** 2))[::-1]
-        appx_rank = np.argmax(frob_K < block_tol)
+        
+        # np.argmax finds the first entry for which the tolerance is greater
```

Comment:
I can see that this makes a world of difference.  This comment is fantastic because it clearly explains why this is necessary!